Source/gmenu.cpp

```diff
@@ -126,38 +126,32 @@ void gmenu_call_proc(TMenuItem *pItem, void(*gmFunc)(TMenuItem *))
 		}
 	}
 	sgpCurrItem = &v2[v3 - 1];
-	gmenu_up_down(1);
+	gmenu_up_down(TRUE);
 }
 // 525740: using guessed type int PauseMode;
 // 634464: using guessed type char byte_634464;
 // 63448C: using guessed type int dword_63448C;
 
-void gmenu_up_down(int a1)
+void gmenu_up_down(BOOL isDown)
 {
-	TMenuItem *v1; // eax
-	int v2;        // edi
+	int i;
 
-	v1 = sgpCurrItem;
 	if (sgpCurrItem) {
 		byte_634464 = 0;
-		v2 = dword_63448C;
-		while (v2) {
-			--v2;
-			if (a1) {
-				++v1;
-				sgpCurrItem = v1;
-				if (v1->fnMenu)
-					goto LABEL_10;
-				v1 = dword_634480;
+		i = dword_63448C;
+		while (i) {
+			i--;
+			if (isDown) {
+				++sgpCurrItem;
+				if (!sgpCurrItem->fnMenu)
+					sgpCurrItem = dword_634480;
 			} else {
-				if (v1 == dword_634480)
-					v1 = &dword_634480[dword_63448C];
-				--v1;
+				if (sgpCurrItem == dword_634480)
+					sgpCurrItem = &dword_634480[dword_63448C];
+				--sgpCurrItem;

```

Comment:
Maybe use `sgpCurrItem++` and `sgpCurrItem--` as usual ?

Reply:
lol @AJenbo did we react at the same time? xD